homeassistant/helpers/entity_platform.py

```diff
@@ -53,6 +54,18 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def async_shutdown(hass: HomeAssistant) -> None:
```

Comment:
EntityComponent is responsible for initializing and tearing down entity platforms. Let's move this logic into `EntityComponent.async_setup`

Reply:
Now we could have a generic shutdown listener for entity component that iterates over all entity components, which forward it to all entity platforms.

Reply:
Moved to `async_setup`